src/menu-bar.tsx

```diff
@@ -0,0 +1,85 @@
+/** @jsxImportSource minimal-view */
+
+import {  view, web } from 'minimal-view'
+import {App} from './app'
+
+
+export const MenuBar = web(view('menuBar',
+  class props {
+    app!: typeof App.Context
+  },
+  class local {
+
+  },
+  function actions({ $, fns, fn }) {
+    return fns(new class actions { 
+      setBPM = function(this: HTMLInputElement ) { 
+        $.app.audio.$.bpm = this.valueAsNumber
+        console.log($.app.audio.$.bpm)
+      }
+
+      save = function(){
```

Comment:
this should be onSave prop not an action, we pass it from above `<MenuBar onSave={...} />`